NWN.Permissions/src/ChatCommands/UserAddPermissionCommand.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using System.Collections.Generic;
+using Anvil.API;
+using Anvil.Services;
+using Jorteck.ChatTools;
+
+namespace Jorteck.Permissions
+{
+  [ServiceBinding(typeof(IChatCommand))]
+  internal class UserAddPermissionCommand : IChatCommand
+  {
+    private readonly ConfigService configService;
+
+    public string Command => configService.GetFullChatCommand("user addpermission");
+    public string[] Aliases => null;
+
+    public Dictionary<string, object> UserData { get; } = new Dictionary<string, object>
+    {
+      [PermissionConstants.ChatUserDataKey] = PermissionConstants.UserAddPermission,
+    };
+
+    public int? ArgCount => 1;
+    public string Description => "Grants a permission to a user.";
+
+    public CommandUsage[] Usages { get; } =
+    {
+      new CommandUsage("<permission_name>", "Grant the specified permission to the target user."),
+    };
+
+    public UserAddPermissionCommand(ConfigService configService)
+    {
+      this.configService = configService;
+    }
+
+    public void ProcessCommand(NwPlayer caller, IReadOnlyList<string> args)
+    {
+      string permission = args[0];
+      caller.EnterTargetMode(eventData => eventData.ProcessOnValidPCTargetBy(caller, AddUserPermissionToTarget(permission)));
+    }
+
+    private Action<NwPlayer, NwPlayer> AddUserPermissionToTarget(string permission)
+    {
+      return (NwPlayer target, NwPlayer caller) =>
+      {
```

Comment:
Nested functions adds some complexity, but solves passing arguments to actions without having to define a multitude of overloads.

Reply:
Should be able to reduce the nesting with the player extension in my other comment